src/components/HomePage/SearchResultsArea/Offer/OfferContent.js

```diff
@@ -1,22 +1,89 @@
-import React from "react";
+import React, { useState } from "react";
 import PropTypes from "prop-types";
 
-import { Typography } from "@material-ui/core";
+import { Avatar, Typography, Link, Grid, Tooltip, Divider } from "@material-ui/core";
 import Offer from "./Offer";
+import OfferContentListItem from "./OfferContentListItem";
 
 import LoadingMagnifyGlass from "./loading_magnify_glass_svg";
 import { useDesktop } from "../../../../utils/media-queries";
+import useSession from "../../../../hooks/useSession";
 
 import useSearchResultsWidgetStyles from "../SearchResultsWidget/searchResultsWidgetStyles";
 import LOADING_MESSAGES from "./offerLoadingMessages";
-import { DateRange, LocationCity } from "@material-ui/icons";
-import { format, parseISO } from "date-fns";
+import {
+    DateRange, LocationOn, Work, Schedule, MonetizationOn, MoneyOff,
+    FindInPage, Info,
+} from "@material-ui/icons";
+import { format, parseISO, formatDistanceToNowStrict } from "date-fns";
+import ChipList from "./ChipList";
+import AdminReasonModal from "./AdminReasonModal";
+
+import JOB_OPTIONS from "../../SearchArea/AdvancedSearch/JobOptions";
+
+import { addSnackbar } from "../../../../actions/notificationActions";
+
+import { connect } from "react-redux";
+import { getHumanError } from "../../../../utils/offer/OfferUtils";
+import { Skeleton } from "@material-ui/lab";
+import OfferDescription from "./OfferDescription";
+import OfferVisibilityOptions from "./OfferVisibilityOptions";
 
 const getRandomOngoingSearchMessage = () => LOADING_MESSAGES[Math.floor(Math.random() * LOADING_MESSAGES.length)];
 
-const OfferContent = ({ offer, loading }) => {
-    const classes = useSearchResultsWidgetStyles({ isMobile: !useDesktop() });
-    if (loading) {
+const defaultLogo = require("./default_icon.svg");
+
+const OfferContent = ({ addSnackbar, offer, loading, isPage }) => {
+
+    const { data, isValidating, error, isLoggedIn } = useSession();
+    const sessionData = (!isValidating && !error && isLoggedIn) ? data : null;
+    const isMobile = !useDesktop();
+    const classes = useSearchResultsWidgetStyles({ isMobile, isPage, loading });
+
+    const [visibilityState, setVisibilityState] = useState({
+        isVisible: undefined,
+        isDisabled: undefined,
+        isBlocked: undefined,
+    });
+    visibilityState.isDisabled = offer?.isHidden && offer?.hiddenReason === "ADMIN_REQUEST";
+    visibilityState.isVisible = !offer?.isHidden;
+    visibilityState.isBlocked = offer?.isHidden && offer?.hiddenReason === "COMPANY_BLOCKED";
```

Comment:
This should be computed inside a useEffect, that runs whenever the offer.isHidden or offer.hiddenReason change